linode/manager
# refactor: [M3-6343] - MUI v5 Migration - Grid v2 for Features

## Description 📝
Migrating to MUI Grid v2. This covers everything in the `src/features` directory (excluding Longview which is forthcoming) and any additional files that needed to be cleaned up.

> **Note**: In several places, you _MAY_ notice slight padding variations due to overly excessive `Grid` usage. In places like this, I made my best judgment and consulted with UX/Design. You'll notice in several places the use of `padding: 0` on `Grid` elements. This was to match the existing design as much as possible until we get a chance to revisit these areas. Feel free to highlight them, and I'll comment!

## Major Changes 🔄 
- `MainContent.tsx` - You'll notice _slightly_ more padding for the main content container. This is because the page is actually respecting the original padding of the container now.
- Changed `Grid` import from `src/components/Grid` to `@mui/material/Unstable_Grid2`. 
   - The `item` attribute is no longer needed for `Grid`.
   - Most `Grid` elements with the `container` attribute needed a `spacing` attribute as part of the change
- `updateFor` has been removed off the Grid component. There's no need to micromanage how/when a component should re-render like this anymore.

### Refactored
- `LinodeEntityDetail.tsx` - Very large file, did some major refactoring here to get things looking like they did before. Check all breakpoints. You'll notice that `DNS Resolvers` has been updated to use CSS Grid and its layout on smaller breakpoints is slightly different. UX approves 👍 
- `DNSResolvers.tsx` - Simplified code and removed `makeStyles`
- `NodeBalancerDetail.tsx` - Using `LandingHeader` and removed `makeStyles`
- `RenderNotification.tsx` - No need for Grid, this was an easy refactor
- `useFormattedNotifications.tsx` - Due to changes in `RenderNotification` this needed a slight refactor
- `LinodeAdvancedConfigurationsPanel.tsx` - Didn't seem like the styles here were being used
- `AlertSection.tsx` - Instead of forcing changes to work, I saw refactoring was more simple.

### Replaced Grid w/ Box component
There was no need for a `Grid` components in these files, so I swapped them with `Box` or removed the `Grid` completely.
- `ExpandableTicketPanel.tsx` - Removed lots of extra `Grid` components in this file
- `AutoEnroll.tsx`
- `DatabaseSummaryConnectionDetails.tsx`
- `FirewallRuleTable.tsx`
- `SupportSearchLanding.tsx`
- `ImageSelect.tsx`
- `KubeConfigDisplay.tsx`
- `NodeBalancerConfigurations.tsx`
- `TablesPanel.tsx` - This is under NodeBalancers
- `VolumeTableRow.tsx`

### Deleted
- `features/Domains/DomainsTableWrapper.tsx` - This didn't seem to be used

## How to test
1. These changes touch the core features we offer, so a page by page scan is required
2. Go to https://cloud.linode.com/ and compare side by side with local
3. There are slight changes as noted above and you may notice slight padding changes. Anything agregious, just flag in this PR and I'll take a look at.
4. Start with larger breakpoints and work your way down.

<details>
<summary>Where to view changes? Open me! 👇 </summary>
<br>

| Component    | Where to view or Desc? | Screenshot |
| ----------- | ----------- | ----------- |
| AutoEnroll <br /> http://localhost:3000/account/settings | Click on "enable now" link to open drawer. If you're a "managed" user, you may not be able to see this. | |
| DatabaseSummaryConnectionDetails <br /> http://localhost:3000/databases/postgresql/{ID}/summary | | ![Screen Shot 2023-04-12 at 10 18 49 AM](https://user-images.githubusercontent.com/125309814/231487000-2685ad0e-0493-444b-ab32-987535e58c35.png)|
| Firewalls Rules <br /> http://localhost:3000/firewalls/{ID} | | ![Screen Shot 2023-04-12 at 10 21 19 AM](https://user-images.githubusercontent.com/125309814/231487552-178d83a1-4192-4914-8cc1-bc65749ea6d2.png) |
| ImageSelect <br /> http://localhost:3000/stackscripts/create | | ![Screen Shot 2023-04-12 at 10 28 02 AM](https://user-images.githubusercontent.com/125309814/231489504-c65aa943-87c6-4385-a4a3-b353b487159c.png)|
| KubeConfigDisplay <br /> http://localhost:3000/kubernetes/clusters/{ID}/summary | | ![Screen Shot 2023-04-12 at 10 30 27 AM](https://user-images.githubusercontent.com/125309814/231490385-1a1a265c-afdd-4aa2-80ef-d65aa8c0441d.png)|
| NodeBalancerConfigurations  <br/> http://localhost:3000/nodebalancers/{ID}/configurations | Configurations tab, lots of refactoring here | ![Screen Shot 2023-04-12 at 10 33 39 AM](https://user-images.githubusercontent.com/125309814/231491148-86bfa139-b67c-44e0-920d-ebda305ff3be.png) |
| Linode Summary View <br /> http://localhost:3000/linodes?view=grid | Tags were using a 1400 breakpoint which is not a supported breakpoint. These will now change layout at < 1280 | ![Screen Shot 2023-04-12 at 9 20 59 AM](https://user-images.githubusercontent.com/125309814/231472242-f31a63c0-acc8-474f-80ba-f8092e84f8af.png)|
| Linode Detail <br /> http://localhost:3000/linodes/{ID}/networking | DNS Resolvers has been refactored | ![Screen Shot 2023-04-12 at 9 35 58 AM](https://user-images.githubusercontent.com/125309814/231475050-b073723a-db04-41b9-8ee2-7fca4b3880d7.png)|
| Notifications <br /> http://localhost:3000/profile/settings | Use mock service worker > open events dropdown | ![Screen Shot 2023-04-12 at 9 57 00 AM](https://user-images.githubusercontent.com/125309814/231480854-d6be4879-a661-4bbf-af48-20519c850afa.png) |
| ExpandableTicketPanel <br /> http://localhost:3000/support/tickets | Use MSW | ![Screen Shot 2023-04-12 at 10 02 05 AM](https://user-images.githubusercontent.com/125309814/231482161-5a78c4b8-0736-4a32-8a84-6a9c85eb4654.png) |
| Entity Transfer Create <br /> http://localhost:3000/account/service-transfers/create <br /> http://localhost:3000/account/service-transfers | | |
|Firewalls Linodes <br/> http://localhost:3000/firewalls/{ID}/linodes|| ![Screen Shot 2023-04-12 at 10 09 24 AM](https://user-images.githubusercontent.com/125309814/231483999-b513fbe8-118f-41a5-8675-81543ef9e73b.png)|
| Search Results <br /> http://localhost:3000/support/search/?query=test | | ![Screen Shot 2023-04-12 at 10 11 00 AM](https://user-images.githubusercontent.com/125309814/231484491-e0915e93-7aea-415c-a73e-b2048efb9be5.png)|
| User Menu| Top navigation user menu |  ![Screen Shot 2023-04-12 at 11 29 52 AM](https://user-images.githubusercontent.com/125309814/231507892-12ac8adb-67d7-4392-a34d-bf4b37a64eaf.png)|
</details>

## QA Checklist
Tag your name next to the ones you've QA'd
- [x] Linodes
- [x] Volumes
- [x] NodeBalancers
- [x] Firewalls
- [x] Stackscripts
- [x] Domains
- [x] Databases
- [x] Kubernetes
- [x] Object Storage
- [x] Marketplace - Shouldn't be any changes here
- [x] Account
- [x] Help & Support
